libdnf/rpm/package_query.cpp

```diff
@@ -1869,7 +1869,7 @@ void PackageQuery::filter_supplements(const PackageSet & package_set, libdnf::sa
 }
 
 void PackageQuery::filter_advisories(
-    const libdnf::advisory::AdvisoryQuery & advisory_query, libdnf::sack::QueryCmp cmp_type) {
+    const std::vector<libdnf::advisory::AdvisoryPackage> & adv_pkgs, libdnf::sack::QueryCmp cmp_type) {
```

Comment:
There is one disadvantage of the approach. If the method internally requires sorted vector then it must resort it to ensure  the proper order. Somehow I am not sure whether it is a good move. From advisory query the method can generate the list if the methot requires it. Advisory query has or can have also some special feature related to advisory that are not defined in vector like search in container.

Reply:
I have split the `filter_advisories()` into two. The public one still accepts `advisory_query` and then there is a private `filter_sorted_advisory_pkgs()` that we use internally from `goal.cpp` that requires the sorted vector.

If you prefer I could make the `filter_sorted_advisory_pkgs()` less general and change it to accept also `advisory_query` but it would in addition select only the latest `AdvisoryPackages`.